### PR TITLE
Make some WMAgent/WorkQueue metrics MonIT-friendly

### LINKED ISSUE #8480: Push worker threads execution time to MONIT IT
Since we now collect this data and persist it in the relational database, it should be pretty straight forward to publish how long each component and their threads take per cycle.
This is going to be more like a developers metric/view, but I think it's quite useful.

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes https://github.com/dmwm/WMCore/issues/8480 and #8928
Requires deployment changes: https://github.com/dmwm/deployment/pull/696
Changes are:
* updated views removing backward compatible changes done in #7554
* no longer dump those monitoring metrics in a local json file
* make sure some keys like job type, schedd status, workqueue element status are always present in the output (for both WMStats and MonIT)
* moved BossAir DAO under BossAir instead of WMBS (RunJobByStatus)
* changed the data structure for a few metrics uploaded to MonIT (affecting global_workqueue too)
* changed the data structure for possible and unique jobs
* take AMQ credentials from the agent secrets file
* fetch worker thread info from wm_workers and push those to MonIT
* renamed WMAgent document `type` pushed to MonIT
* removed reqmgr (one) bits from the deployment script

TODO: write a nice wiki with every single doc type in MonIT and the meaning of each field.

For the record, an empty agent creates around 125 docs to be uploaded to AMQ every cycle.

### CODE REVIEW

**Comment by amaltaro** [comment]
test this please

File: src/python/WMCore/BossAir/MySQL/RunJobByStatus.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+all sched_status are always present in the output.
+"""
+from __future__ import print_function, division
+
+from WMCore.Database.DBFormatter import DBFormatter
+
+
+class RunJobByStatus(DBFormatter):
+    sql = """
+          SELECT bl_status.name AS sched_status, count(bl_runjob.sched_status) AS count, bl_runjob.status
```
Comment (ticoann) [inline src/python/WMCore/BossAir/MySQL/RunJobByStatus.py]: Alan, does sched_status point to actual schedd states, IDLE, Removed, Held, etc or mapped states Pending, Running, etc?

Comment (amaltaro) [reply src/python/WMCore/BossAir/MySQL/RunJobByStatus.py]: Yes, I figured it would to more good than harm.

**Comment by amaltaro** [comment]
@ticoann sorry for being late with this (and piling all this work into the same PR!), hopefully I won't push in anything else but pylint fixes.

I did ran some tests in my VM and it looks good. However it's sort of a mess by now and the last couple of commits haven't been tested yet. 

Anyways, can I ask you to start looking into it? I know it's going to be painful without having it documented, I shall have that done by tomorrow.

**Comment by amaltaro** [comment]
@ticoann after deploying an agent from scratch, I found and fixed 3 bugs and now everything seems to be working fine (including the deployment part!).
I was also able to find data in Kibana (wmagentqa index pattern) with the following query, e.g.:
`data.payload.agent_url:vocms0192.cern.ch AND data.payload.type:wma_health_info`

FYI @vkuznet @stiegerb , we'll document the field names/meaning and write example of document types and data structure uploaded to AMQ/MonIT. However, feel free to point out better data structure where possible.
FYI @vytjan we can eventually add T0 specifics on top of it.

File: src/python/WMCore/Services/WorkQueue/WorkQueue.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 from WMCore.Database.CMSCouch import CouchServer, CouchConflictError
 from WMCore.Lexicon import splitCouchServiceURL
 from WMCore.WMSpec.WMWorkload import WMWorkloadHelper
+from WMCore.WorkQueue.DataStructs.WorkQueueElement import STATES
```
Comment (amaltaro) [inline src/python/WMCore/Services/WorkQueue/WorkQueue.py]: it looks like this import causes import problems when it's used by ReqMgr2. I'll have to add `WMCore.WorkQueue.DataStructs+` as a dependency for reqmgr2

**Comment by ticoann** [comment]
Test this please

**Review by ticoann — APPROVED** [review]

**Comment by ticoann** [comment]
It seems something is wrong with Jenkins. Since this is tested in vm. I am merging it. Thanks Alan

**Pull request merged**